pkg/challenges/challenges.go

```diff
@@ -69,6 +70,78 @@ type ChallengeResult struct {
 	AdditionalInfo map[AdditionalInfo]string
 }
 
+func (cr *ChallengeResult) Name(context.Context) string {
+	return cr.Value
+}
+
+func (cr *ChallengeResult) Embed(ctx context.Context, cert *x509.Certificate) error {
+	switch cr.TypeVal {
+	case EmailValue:
+		cert.EmailAddresses = []string{cr.Value}
+	case SpiffeValue:
+		challengeURL, err := url.Parse(cr.Value)
+		if err != nil {
+			return err
+		}
+		cert.URIs = []*url.URL{challengeURL}
+	case GithubWorkflowValue:
+		jobWorkflowURI, err := url.Parse(cr.Value)
+		if err != nil {
+			return err
+		}
+		cert.URIs = []*url.URL{jobWorkflowURI}
+	case KubernetesValue:
+		k8sURI, err := url.Parse(cr.Value)
+		if err != nil {
+			return err
+		}
+		cert.URIs = []*url.URL{k8sURI}
+	case URIValue:
+		subjectURI, err := url.Parse(cr.Value)
+		if err != nil {
+			return err
+		}
+		cert.URIs = []*url.URL{subjectURI}
+	case UsernameValue:
+		cert.EmailAddresses = []string{cr.Value}
+	}
+
+	exts := x509ca.Extensions{
+		Issuer: cr.Issuer,
+	}
+	if cr.TypeVal == GithubWorkflowValue {
+		var ok bool
+		exts.GithubWorkflowTrigger, ok = cr.AdditionalInfo[GithubWorkflowTrigger]
+		if !ok {
+			return errors.New("x509ca: github workflow missing trigger claim")
```

Comment:
can you drop "x509ca" from each of these error messages? they don't have any value